### ISSUE TITLE
ansible-lint 6.14.6 fails with 'internal-error:' if vault is not set (in some edge cases)

### ISSUE THREAD

**Issue by hille721** [ansible/ansible-lint#3308]
##### Summary

In my CI ansible-lint fails since the last release with:

```
internal-error: Unexpected error code 1 from execution of: ansible-playbook -i localhost, --syntax-check /tmp/playwab_d5ke.yml
```

I don't have my ansible vault configured there as ansible-lint was working well without it till the last release.

##### Issue Type

- Bug Report

##### OS / ENVIRONMENT

```console (paste below)
ansible-lint --version
ansible-lint 6.14.6 using ansible 2.14.4
```

- ansible installation method: pip
- ansible-lint installation method: pip

##### STEPS TO REPRODUCE

Assume following `test` role:

```console (paste below)
roles/test/
├── defaults
│   └── main
│       ├── main.yml
│       └── vault.yml
└── tasks
    └── main.yml
```

`defaults/main/vault.yml` is a file vault encrypted file which only contains:

```yml
_vault_variable: test
```

`defaults/main/main.yml` contains following:

```yml
vault_variable: "{{ _vault_variable }}"
```

and `tasks/main.yml` simple:

```yml
- name: Vault
  ansible.builtin.debug:
    var: vault_variable
```

##### Desired Behavior

ansible-lint should run successful as before and should not require the vault

##### Actual Behavior

```paste below
$ ansible-lint 
WARNING  Listing 1 violation(s) that are fatal
internal-error: Unexpected error code 1 from execution of: ansible-playbook -i localhost, --syntax-check /tmp/play85_oxizy.yml
roles/test:1 ERROR! Attempting to decrypt but no vault secrets found

Read documentation for instructions on how to ignore specific rule violations.

              Rule Violation Summary               
 count tag            profile rule associated tags 
     1 internal-error min     core                 

Failed after : 1 failure(s), 0 warning(s) on 28 files.
```

[minimum complete verifiable example]: http://stackoverflow.com/help/mcve
